### PR TITLE
Rework ILM build to separate integration tests

### PR DESCRIPTION
**Pull request by AthenaEryma**
Having integration tests separated from the unit tests in the qa
directory works much more smoothly with our testing infrastructure,
matches what other plugins do, and tests in a more "real" deployment
scenario by having all plugins installed.

This should fix the current build problems with the `index-lifecycle` branch.

### CODE REVIEW

File: x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleInitialisationTests.java [hunk 1]
```diff
@@ -72,7 +72,7 @@
 import static org.hamcrest.core.IsNull.nullValue;
 
 @ESIntegTestCase.ClusterScope(scope = Scope.TEST, numDataNodes = 0)
-public class IndexLifecycleInitialisationIT extends ESIntegTestCase {
+public class IndexLifecycleInitialisationTests extends ESIntegTestCase {
```
Comment (gwbrown) [inline x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleInitialisationTests.java]: This is necessary to run this class with `integTest.enabled = false` - several other plugins also do this.

Comment (nik9000) [reply x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleInitialisationTests.java]: It is how other plugins do it, yeah. There isn't a clear definition of "correct" here.

**Review by nik9000 — APPROVED** [review]
LGTM

**Review by talevy — APPROVED** [review]
LGTM if tests are 💚

**Pull request merged**